core/src/main/java/hudson/model/User.java

```diff
@@ -713,8 +714,9 @@ public static boolean isIdOrFullnameAllowed(@CheckForNull String id) {
         if (id == null || StringUtils.isBlank(id)) {
             return false;
         }
+        final String trimmedId = id.trim();
         for (String invalidId : ILLEGAL_PERSISTED_USERNAMES) {
-            if (id.equalsIgnoreCase(invalidId))
+            if (trimmedId.equalsIgnoreCase(invalidId))
```

Comment:
Why? I was under the impression that String.equalsIgnoreCase is safe even for different locales.